### DISCUSSION TITLE
dnscore : custom zone file does not work

### DISCUSSION THREAD

**Discussion by klykovn** [coredns/coredns#6116] [Q&A]
Hi everyone. 
According to steps described in https://coredns.io/2017/05/08/custom-dns-entries-for-kubernetes/
I've added my own zone file to the coredns config, but nothing works.
Here is the ConfigMap: 
apiVersion: v1
data:
  Corefile: |
    .:53 {
        errors
        log
        health {
            lameduck 5s
        }
        ready

        kubernetes cluster.local in-addr.arpa ip6.arpa {
          pods insecure
          fallthrough in-addr.arpa ip6.arpa
        }

        prometheus :9153
        file /etc/coredns/test.db blue.test

        forward . /etc/resolv.conf {
          prefer_udp
        }
        cache 30

        loop
        reload
        loadbalance
    }
  test.db: |
    ; blue.test zone
    blue.test.                   IN          SOA         sns.dns.icann.org.  noc.dns.icann.org. 2019101701 7200 3600 1209600 3600
    blue.test.                   IN          NS          b.iana-servers.net.
    blue.test.                   IN          NS          b.iana-servers.net.
    blue.test.                   IN          A           127.0.0.1
    something.blue.test.         IN          CNAME       myservice.svc.cluster.local.
kind: ConfigMap

The test.db file is also present in the deployment volume.  : 
 volumes:
      - configMap:
          defaultMode: 420
          items:
          - key: Corefile
            path: Corefile
          - key: test.db
            path: test.db
          name: coredns
        name: config-volume

# kubectl run -it --rm --restart=Never --image=infoblox/dnstools:latest dnstools
If you don't see a command prompt, try pressing enter.
dnstools# nslookup blue.test

Server:         169.254.25.10
Address:        169.254.25.10#53

** server can't find blue.test: NXDOMAIN

dnstools# exit
pod "dnstools" deleted
pod default/dnstools terminated (Error)

# kubectl logs coredns-7755d747b7-lscf2 -n kube-system
.:53
[INFO] plugin/reload: Running configuration MD5 = 41b5b62b20e50d6c0ee15afe1226b804
CoreDNS-1.8.6
linux/amd64, go1.17.1, 13a9191

Address:        169.254.25.10#53

** server can't find blue.test: NXDOMAIN

[INFO] 192.168.0.16:49839 - 37633 "A IN blue.test.default.svc.cluster.local. tcp 53 false 65535" NXDOMAIN qr,aa,rd 146 0.000207714s
[INFO] 192.168.0.16:49839 - 37166 "A IN blue.test.svc.cluster.local. tcp 45 false 65535" NXDOMAIN qr,aa,rd 138 0.000208014s
[INFO] 192.168.0.16:49839 - 26029 "A IN blue.test.cluster.local. tcp 41 false 65535" NXDOMAIN qr,aa,rd 134 0.000091629s

What I'm doing wrong ? 
I would be grateful for help.

**Comment by chrisohaver** [comment]
If you have node local dns cache deployed, the file plugin needs to be instead configured in the node local dns cache instances of coredns.

**Comment by klykovn** [comment]
Yes, I have it.Thank you!  I will try and write about the results.

**Comment by klykovn** [comment]
It works! Thanks a lot! 
:)
